pkg/model/elbv2/target_group.go

```diff
@@ -0,0 +1,128 @@
+package elbv2
+
+import (
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/aws-alb-ingress-controller/pkg/model/core"
+)
+
+var _ core.Resource = &TargetGroup{}
+
+// TargetGroup represents a ELBV2 TargetGroup
+type TargetGroup struct {
+	// resource id
+	id string
+
+	// desired state of TargetGroup
+	spec TargetGroupSpec `json:"spec"`
+
+	// observed state of TargetGroup
+	// +optional
+	status *TargetGroupStatus `json:"status,omitempty"`
+}
+
+// NewLoadBalancer constructs new LoadBalancer resource.
+func NewTargetGroup(stack core.Stack, id string, spec TargetGroupSpec) *TargetGroup {
+	tg := &TargetGroup{
+		id:     id,
+		spec:   spec,
+		status: nil,
+	}
+	stack.AddResource(tg)
+	return tg
+}
+
+// ID returns resource's ID within stack.
+func (tg *TargetGroup) ID() string {
+	return tg.id
+}
+
+type TargetType string
+
+const (
+	TargetTypeInstance TargetType = "instance"
+	TargetTypeIP                  = "ip"
+)
+
+// Information to use when checking for a successful response from a target.
+type HealthCheckMatcher struct {
+	// The HTTP codes.
+	HTTPCode string `json:"httpCode"`
+}
+
+// Configuration for TargetGroup's HealthCheck.
+type TargetGroupHealthCheckConfig struct {
```

Comment:
Any reason why all fields are pointers in this struct?

Reply:
1. since they are optional fields in aws API as well. 
so i want to follow this principle around the model: if the type is optional in AWS API, make it optional, and the semantic is it will default to aws's default value server-side. and during reconcile, the settings won't be touched.

2. even the HealthCheck is required for ALB, i make `HealthCheckConfig *TargetGroupHealthCheckConfig` a pointer. So in the future, we can support lambda targets via annotations :D